src/engine/board.ts

```diff
@@ -7,8 +7,8 @@ export default class Board {
     public currentPlayer: Player;
     private readonly board: (Piece | undefined)[][];
 
-    public constructor() {
-        this.currentPlayer = Player.WHITE;
+    public constructor(currentPlayer?: Player) {
+        this.currentPlayer = currentPlayer ? currentPlayer : Player.WHITE;
```

Comment:
I don't think you wrote this but just FYI as it's useful this can be simplified to ` currentPlayer || Player.WHITE`
`X || Y `means if X then X else Y

Reply:
We didn't write this, unsure of the difference??

Reply:
functionally they are the exact same just writing` X || Y` over `X ? X : Y` is more concise

Reply:
Fixed - was wondering if it's clearer to write X || Y as its not immediately clear to someone that X and Y are bools so they are equivalent

Reply:
Fixed